backtrader/brokers/ccxtbroker.py

```diff
@@ -71,18 +74,27 @@ def get_notification(self):
     def notify(self, order):
         self.notifs.put(order)
 
-    def getposition(self, data):
-        currency = data.symbol.split('/')[0]
-        return self.store.getposition(currency)
+    def getposition(self, data, clone=True):
+        # return self.o.getposition(data._dataname, clone=clone)
+        pos = self.positions[data._dataname]
+        if clone:
```

Comment:
What if we already have position when algo started? How we're going to report it?

Reply:
Hi Ed, 
  Yes - There is still work to do. However, this does get us a step closer I think. 

Regarding the question. It is a good one. The golden sample, aka Daniels own Oanda integration tackles this on init

```
if self.p.use_positions:
            for p in self.o.get_positions():
                print('position for instrument:', p['instrument'])
                is_sell = p['side'] == 'sell'
                size = p['units']
                if is_sell:
                    size = -size
                price = p['avgPrice']
                self.positions[p['instrument']] = Position(size, price)
```

This will get the open positions when we start. Naturally we would need to tweak this to work with the CCXT store.

Reply:
Hi Ed, 
  Just to expand on above a bit further. The original code for getting positions also would not allow Backtrader to be aware of open positions. This is because the `getposition` method in the store only returns the account value. It needs to return the position size for a particular instrument and pass it to the positions dictionary.

Then we can use the `if not self.position` check in Backtrader.   

Snippet from the CCXT store:
```
@retry
    def getposition(self, currency):
        return self.getvalue(currency)
```

The problem I was having yesterday was that I could not find a good unified method for returning open positions in the CCXT library. This is where I started playing with the parse trades method below. I was thinking to get a list of trades and see if any of them were open. I did not resolve this item though.